src/core/imap/MCIMAPSession.cpp

```diff
@@ -3743,6 +3743,7 @@ void IMAPSession::storeLabels(String * folder, bool identifier_is_uid, IndexSet
             return;
         }
     }
```

Comment:
Could change the "return"s above to "goto release" too? It would be a great mem leak fix :)